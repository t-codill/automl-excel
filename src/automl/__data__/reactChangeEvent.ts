// tslint:disable: no-any
export const reactChangeEvent: React.ChangeEvent<any> = {
    target: {}
} as unknown as React.ChangeEvent<any>;
