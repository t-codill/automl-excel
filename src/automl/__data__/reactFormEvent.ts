// tslint:disable: no-any
export const reactFormEvent: React.FormEvent<any> = {
    preventDefault(): void { return; }
} as unknown as React.FormEvent<any>;
