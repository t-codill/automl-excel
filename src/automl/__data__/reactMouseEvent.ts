// tslint:disable-next-line: no-any
export const reactMouseEvent = {} as unknown as React.MouseEvent<any>;
