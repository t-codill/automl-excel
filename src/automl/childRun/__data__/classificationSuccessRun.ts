import { NotNullableProperties } from "../../common/utils/NotNullableProperties";
import { IChildRunData } from "../ChildRun";

export const classificationSuccessRun: NotNullableProperties<IChildRunData> = {
    run: {
        runId: "AutoML_001",
        parentRunId: "AutoML_000",
        status: "Completed",
        startTimeUtc: new Date("2019-02-01T08:20:00.000Z"),
        endTimeUtc: new Date("2019-02-01T08:20:30.000Z"),
        properties: {
            iteration: "9",
            run_algorithm: "Ensemble",
            run_preprocessor: "rpp1"
        }
    },
    runMetrics: {
        scalarMet1: "1",
        scalarMet2: "2",
        scalarMet3: "3",
        accuracy_table: {},
        confusion_matrix: {}
    },
    modelUri: "sampleModelUri",
    experimentName: "classificationSuccess"
};
