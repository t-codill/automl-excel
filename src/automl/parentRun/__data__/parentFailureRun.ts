import { NotNullableProperties } from "../../common/utils/NotNullableProperties";
import { IParentRunData } from "../ParentRun";

export const parentFailureRun: NotNullableProperties<IParentRunData> = {
    run: {
        runId: "AutoML_000",
        parentRunId: undefined,
        status: "Completed",
        startTimeUtc: new Date("2019-02-01T08:20:00.000Z"),
        endTimeUtc: new Date("2019-02-01T08:30:00.000Z"),
        properties: {
            AMLSettingsJsonString: "",
            primary_metric: "pm",
            errors: "Setup iteration failed: \\n \t\t\t\t\texception: \\n \t\t\t\t\t\tUnforseenConsequence('Invalid subject in the test chamber') \\n"
        }
    },
    experimentName: "Test",
    childRuns: [],
    childRunMetrics: [
        {
            spearman_correlation: 0.877770827248834
        }
    ]
};
