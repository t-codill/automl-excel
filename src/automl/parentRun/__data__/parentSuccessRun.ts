import { NotNullableProperties } from "../../common/utils/NotNullableProperties";
import { IParentRunData } from "../ParentRun";

export const parentSuccessRunAMLSettings = {
  primary_metric: "AUC_weighted",
  data_script: null,
  compute_target: "local",
  n_cross_validations: 3,
  max_cores_per_iteration: 1,
  blacklist_algos: [],
  exit_score: 100,
  name: "my_experiment",
  task_type: "classification"
};
export const parentSuccessRun: NotNullableProperties<IParentRunData> = {
  run: {
    runId: "AutoML_000",
    parentRunId: undefined,
    status: "Completed",
    startTimeUtc: new Date("2019-02-01T08:20:00.000Z"),
    endTimeUtc: new Date("2019-02-01T08:30:00.000Z"),
    properties: {
      AMLSettingsJsonString: JSON.stringify(parentSuccessRunAMLSettings)
    },
    tags: {
      iterations: "2"
    },
  },
  experimentName: "Test",
  childRuns: [
    {
      createdUtc: new Date("2019-02-01T08:20:00.000Z"),
      runId: "AutoML_001",
      parentRunId: "AutoML_000",
      status: "Completed",
      startTimeUtc: new Date("2019-02-01T08:20:00.000Z"),
      endTimeUtc: new Date("2019-02-01T08:25:00.000Z"),
      experimentId: "exp",
      options: {
        generateDataContainerIdIfNotSpecified: true
      },
      name: "AutoML_001",
      properties: {
        runTemplate: "automl_child",
        pipeline_id: "eda663ea8f36e6c07948c99bba9d1becb40991a5",
        run_template: "automl_child",
        run_algorithm: "RandomForest",
        run_preprocessor: "MaxAbsScaler",
        score: "0.977770827248834",
        class_labels: "",
        primary_metric: "spearman_correlation",
      }
    },
    {
      createdUtc: new Date("2019-02-01T08:20:00.000Z"),
      runId: "AutoML_002",
      parentRunId: "AutoML_000",
      status: "Completed",
      startTimeUtc: new Date("2019-02-01T08:20:00.000Z"),
      endTimeUtc: new Date("2019-02-01T08:25:00.000Z"),
      options: {
        generateDataContainerIdIfNotSpecified: true
      },
      name: "AutoML_002",
      properties: {
        runTemplate: "automl_child",
        pipeline_id: "eda663ea8f36e6c07948c99bba9d1becb40991a5",
        run_template: "automl_child",
        run_preprocessor: "MaxAbsScaler",
        score: "0.877770827248834",
        class_labels: "",
        primary_metric: "spearman_correlation",
      }
    },
    {
      createdUtc: new Date("2019-02-01T08:20:00.000Z"),
      runId: "AutoML_003",
      parentRunId: "AutoML_000",
      status: "Canceled",
      startTimeUtc: new Date("2019-02-01T08:20:00.000Z"),
      endTimeUtc: new Date("2019-02-01T08:25:00.000Z"),
      options: {
        generateDataContainerIdIfNotSpecified: true
      },
      name: "AutoML_003",
      properties: {
        runTemplate: "automl_child",
        pipeline_id: "eda663ea8f36e6c07948c99bba9d1becb40991a5",
        run_template: "automl_child",
        score: "0.877770827248834",
        class_labels: "",
        primary_metric: "spearman_correlation",
      }
    },
    {
      createdUtc: new Date("2019-02-01T08:20:00.000Z"),
      runId: "AutoML_004",
      parentRunId: "AutoML_000",
      status: "Completed",
      startTimeUtc: new Date("2019-02-01T08:20:00.000Z"),
      endTimeUtc: new Date("2019-02-01T08:25:00.000Z"),
      options: {
        generateDataContainerIdIfNotSpecified: true
      },
      name: "AutoML_004",
      properties: {
        runTemplate: "automl_child",
        pipeline_id: "eda663ea8f36e6c07948c99bba9d1becb40991a5",
        run_template: "automl_child",
        run_algorithm: "MaxAbsScaler",
        score: "0.827770827248834",
        class_labels: "",
        primary_metric: "spearman_correlation",
      },
    },
    {
      createdUtc: new Date("2019-02-01T08:20:00.000Z"),
      runId: "AutoML_005",
      parentRunId: "AutoML_000",
      status: "Completed",
      startTimeUtc: new Date("2019-02-01T08:20:00.000Z"),
      endTimeUtc: new Date("2019-02-01T08:25:00.000Z"),
      options: {
        generateDataContainerIdIfNotSpecified: true
      },
      name: "AutoML_005",
    }
  ],
  childRunMetrics: [
    {
      spearman_correlation: 0.877770827248834
    }
  ]
};
