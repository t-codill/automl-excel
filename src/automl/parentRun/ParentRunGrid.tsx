import * as React from "react";
import { shouldShowParentRunModel } from "../common/utils/run";
import { GridItem } from "../components/Grid/GridItem";
import { ChildRunIterationTable } from "./ChildRunIterationTable";
import { IParentRunData } from "./ParentRun";
import { ParentRunModel } from "./ParentRunModel";
import { ParentRunSkylineChart } from "./ParentRunSkylineChart";
import { ParentRunSummary } from "./ParentRunSummary";
import { ParentSettings } from "./ParentSettings";

export interface IParentRunGridProps extends IParentRunData {
    onModelDeploy(): void;
}
export class ParentRunGrid extends React.Component<IParentRunGridProps> {
    public readonly render = (): React.ReactNode => {
        const modelVisible = shouldShowParentRunModel(this.props.run && this.props.run.status);
        return (
            <div className="ms-Grid" dir="ltr">
                <div className="ms-Grid-col ms-sm12 ms-md12 ms-lg12 ms-xl9">
                    <GridItem title="Iteration Chart">
                        <ParentRunSkylineChart  {...this.props} />
                    </GridItem>
                    <GridItem title="Iterations">
                        <ChildRunIterationTable  {...this.props} />
                    </GridItem>
                </div>
                <div className="ms-Grid-col ms-sm12 ms-md12 ms-lg12 ms-xl3">
                    {modelVisible && <ParentRunModel {...this.props} />}
                    <GridItem title="Run Summary">
                        <ParentRunSummary run={this.props.run} />
                    </GridItem>
                    <GridItem title="Run Settings">
                        <ParentSettings run={this.props.run} />
                    </GridItem>
                </div>
            </div>
        );
    }
}
