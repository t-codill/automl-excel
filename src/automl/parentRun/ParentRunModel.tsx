import { RunHistoryAPIsModels } from "@vienna/runhistory";
import { isEqual, orderBy } from "lodash";
import * as React from "react";
import { PageNames } from "../common/PageNames";
import { isRunCompleted } from "../common/utils/run";
import { safeParseJson } from "../common/utils/safeParseJson";
import { ModelSection } from "../components/ModelDownloadDeploy/ModelSection";
import { PageLoadingSpinner } from "../components/Progress/PageLoadingSpinner";
import { IParentRunData } from "./ParentRun";
import { IParentRunGridProps } from "./ParentRunGrid";

export interface IParentRunModelState {
    bestRun: RunHistoryAPIsModels.MicrosoftMachineLearningRunHistoryContractsRunDto | undefined;
}

export class ParentRunModel extends React.PureComponent<IParentRunGridProps, IParentRunModelState> {
    constructor(props: IParentRunGridProps) {
        super(props);
        this.state = {
            bestRun: this.getBestRun()
        };
    }

    public async componentDidUpdate(prevProps: IParentRunData): Promise<void> {
        if (isEqual(this.props.childRuns, prevProps.childRuns)) {
            return;
        }
        this.setState({
            bestRun: this.getBestRun()
        });
    }

    public readonly render = (): React.ReactNode => {
        if (!this.state.bestRun || !this.props.run || !this.props.experimentName) {
            return <PageLoadingSpinner />;
        }

        return <ModelSection
            experimentName={this.props.experimentName}
            pageName={PageNames.ParentRun}
            parentRun={this.props.run}
            run={this.state.bestRun}
            onModelDeploy={this.props.onModelDeploy}
        />;
    }

    private readonly getBestRun = () => {
        if (!this.props.childRuns || !this.props.run || !this.props.run.properties) {
            return undefined;
        }
        const min = safeParseJson(this.props.run.properties.AMLSettingsJsonString).metric_operation === "minimize";
        const scoredRuns = this.props.childRuns.map((r) => ({
            ...r,
            runId: r.runId,
            parentRunId: r.parentRunId,
            status: r.status,
            score: r.properties && parseFloat(r.properties.score)
        }));

        const sortedRuns = orderBy(scoredRuns, ["score"], [min ? "asc" : "desc"]);
        for (const bestRun of sortedRuns) {
            if (bestRun.score === undefined || isNaN(bestRun.score) || !isRunCompleted(bestRun)) {
                continue;
            }
            return bestRun;
        }
        return undefined;
    }
}
