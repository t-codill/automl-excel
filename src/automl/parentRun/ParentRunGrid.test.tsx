import { shallow } from "enzyme";
import * as React from "react";
import { parentSuccessRun } from "./__data__/parentSuccessRun";
import { ParentRunGrid } from "./ParentRunGrid";
import { ParentRunModel } from "./ParentRunModel";

describe("Parent Run Grid", () => {
    const props = {
        experimentName: "foo",
        run: {},
        childRuns: [],
        childRunMetrics: [],
        onModelRegister: jest.fn()
    };
    describe("Classification Success", () => {
        it("should render graphs, run metrics and run summary in grid", async () => {
            const tree = shallow(
                <ParentRunGrid {...props} />
            );
            expect(tree)
                .toMatchSnapshot();
        });
    });
    describe("Should render model", () => {
        it("should render graphs, run metrics and run summary in grid", async () => {
            const propsWithRun = { ...props, run: parentSuccessRun.run };
            const tree = shallow(
                <ParentRunGrid {...propsWithRun} />
            );
            expect(tree.find(ParentRunModel)
                .props())
                .toEqual(propsWithRun);
        });
    });
});
