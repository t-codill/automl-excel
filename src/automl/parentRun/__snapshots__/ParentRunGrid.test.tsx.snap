// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parent Run Grid Classification Success should render graphs, run metrics and run summary in grid 1`] = `
<div
  className="ms-Grid"
  dir="ltr"
>
  <div
    className="ms-Grid-col ms-sm12 ms-md12 ms-lg12 ms-xl9"
  >
    <GridItem
      title="Iteration Chart"
    >
      <ParentRunSkylineChart
        childRunMetrics={Array []}
        childRuns={Array []}
        experimentName="foo"
        onModelDeploy={[MockFunction]}
        run={Object {}}
      />
    </GridItem>
    <GridItem
      title="Iterations"
    >
      <ChildRunIterationTable
        childRunMetrics={Array []}
        childRuns={Array []}
        experimentName="foo"
        onModelDeploy={[MockFunction]}
        run={Object {}}
      />
    </GridItem>
  </div>
  <div
    className="ms-Grid-col ms-sm12 ms-md12 ms-lg12 ms-xl3"
  >
    <GridItem
      title="Run Summary"
    >
      <ParentRunSummary
        run={Object {}}
      />
    </GridItem>
    <GridItem
      title="Run Settings"
    >
      <ParentSettings
        run={Object {}}
      />
    </GridItem>
  </div>
</div>
`;
